src/main/scala/io/conduktor/ksm/source/HttpSourceAcl.scala

```diff
@@ -0,0 +1,198 @@
+    this.httpMethod = method
+    this.enableAuth = enableAuth
+    this.serviceAccountName = serviceAccount
+    this.targetAudience = targetAudience
+    this.serviceAccountKey = serviceAccountKey
+  }
+
+  /**
+    * internal config definition for the module
+    */
+  override def configure(config: Config): Unit = {
+    this.uri = config.getString(URL)
+    log.info("URL: {}", this.uri)
+
+    this.parser = config.getString(PARSER)
+    log.info("PARSER: {}", this.parser)
+
+    this.httpMethod = config.getString(METHOD)
+    log.info("HTTP Method: {}", this.httpMethod)
+
+    this.enableAuth = config.getBoolean(ENABLE_AUTH)
+    log.info("Enable Auth: {}", this.enableAuth)
+
+    this.serviceAccountName = config.getString(SERVICE_ACCOUNT)
+    log.info("Service Account: {}", this.serviceAccountName)
+
+    this.targetAudience = config.getString(TARGET_AUDIENCE)
+    log.info("Target Audience: {}", this.targetAudience)
+
+    this.serviceAccountKey = config.getString(SERVICE_ACCOUNT_KEY)
+  }
+
+  /**
+    * Refresh the current view on the external source of truth for Acl
+    * Ideally this function is smart and does not pull the entire external Acl at every iteration
+    * Return `None` if the Source Acls have not changed (usually using metadata).
+    * Return `Some(x)` if the Acls have changed. `x` represents the parsing and parsing errors if any
+    * Note: the first call to this function should never return `None`.
+    *
+    * Kafka Security Manager will not update Acls in Kafka until there are no errors in the result
+    *
+    * @return
+    */
+  override def refresh(): Option[ParsingContext] = {
+
+    val request: Request = new Request(uri)
+    // super important in order to properly fail in case a timeout happens for example
+    request.enableThrowingIOException(true)
+    request.followRedirects(false)
+    request.connectTimeoutMillis(Int.MaxValue)
+    request.readTimeoutMillis(Int.MaxValue)
+
+    if (enableAuth) {
+      val authorization = getIdToken()
+      request.header("Authorization", authorization)
+    }
+
+    // we use this header for the 304
+    lastModified.foreach(header => request.header("If-Modified-Since", header))
+    val response: Response = HTTP.get(request)
+
+    response.status match {
+      case 200 =>
+        lastModified = response.header("Last-Modified")
+
+        Some(
+          ParsingContext(
+            parserRegistry.getParser(this.parser),
+            new BufferedReader(new StringReader(response.textBody))
+          )
+        )
+      case 304 =>
+        None
+      case _ =>
+        // we got an http error so we propagate it
+        log.error("HTTP Status {} in HTTP Source request: {}", response.status, response.asString)
+        throw HTTPException(Some(response.asString), response)
+    }
+  }
+
+  private def getIdToken(): String = {
+    val audience = this.targetAudience
+
+    if (jwtMap.get(audience) != null
+      && expirationMap.get(audience) != null
```

Comment:
`.get` returns an option, it can't be null

Reply:
Done